test/cbmc/proofs/Makefile.common

```diff
@@ -257,12 +257,10 @@ CBMC_FLAG_FLUSH ?= --flush
 
 # CBMC flags used for property checking and coverage checking
 
-CBMCFLAGS += $(CBMC_FLAG_UNWIND) $(CBMC_UNWINDSET) $(CBMC_FLAG_FLUSH)
+CBMCFLAGS += $(CBMC_UNWINDSET) $(CBMC_FLAG_FLUSH)
```

Comment:
Do we not need unwinding anymore? Is it something which happens by default now?

Reply:
`CBMC_FLAG_UNWIND` included the flag `--unwind 1` as a general bound for the CBMC analysis. This is not necessary because you should provide all bounds for the program using `UNWINDSET` OR `EARLY_UNWINDSET`. Take a look at this code snippet from `Makefile.common`:
```Makefile
# CBMC loop unwinding (Normally set in the proof Makefile)
#
# UNWINDSET is a list of pairs of the form foo.1:4 meaning that
# CBMC should unwind loop 1 in function foo no more than 4 times.
# For historical reasons, the number 4 is one more than the number
# of times CBMC actually unwinds the loop.
UNWINDSET ?=

# CBMC early loop unwinding (Normally set in the proof Makefile)
#
# Most users can ignore this variable.
#
# This variable exists to support the use of loop and function
# contracts, two features under development for CBMC.  Checking the
# assigns clause for function contracts and loop invariants currently
# assumes loop-free bodies for loops and functions with contracts
# (possibly after replacing nested loops with their own loop
# contracts).  To satisfy this requirement, it may be necessary to
# unwind some loops before the function contract and loop invariant
# transformations are applied to the goto program.  This variable
# EARLY_UNWINDSET is identical to UNWINDSET, and we assume that the
# loops mentioned in EARLY_UNWINDSET and UNWINDSET are disjoint.
EARLY_UNWINDSET ?=
```

Now that our `Makefile.common` supports [function contracts](https://diffblue.github.io/cbmc/contracts-functions.html), there are some loops introduced by this feature that CBMC itself will unroll during symbolic execution and adding a general loop unwind (i.e., `--unwind 1`) at that stage would cause spurious unwinding assertion violations (i.e., when CBMC complains that bounds are not enough to unroll all loops in the program). Thus, the best solution is to remove the general unwind and rely only on the ones specified by the proof writer.

Reply:
Thanks for that clarification. In case of a new proof, where the proof writer overlooks/forgets a loop, what would be the result? Would that trigger an assertion, or would it just be left without unwinding?

Reply:
Makes sense! 
That was the only concern I had. Approving the PR :)